### PR TITLE
genocides riding runtimes for lzimann and fixes other issues too #23135

### PR DESCRIPTION
**Pull request by Militaires**
boom bam

edit: closes #23411 and #23135

### CODE REVIEW

**Comment by Thunder12345** [comment]
\>New PR
\>Conflicts

File: code/modules/vehicles/speedbike.dm [hunk 1]
```diff
@@ -7,15 +7,14 @@
 	var/image/overlay = null
 
 /obj/vehicle/space/speedbike/buckle_mob()
```
Comment (lzimann) [inline code/modules/vehicles/speedbike.dm]: this also needs the args

Comment (Militaires) [inline code/modules/vehicles/speedbike.dm]: shiet son alright

**Comment by Militaires** [comment]
@Thunder12345 sleight of hand became entertainment and im a big fan of it you just got hary hudini'd

File: code/modules/vehicles/atv.dm [hunk 2]
```diff
@@ -41,7 +41,7 @@
 	turret = new(loc)
 	turret.base = src
 
-/obj/vehicle/atv/turret/buckle_mob()
+/obj/vehicle/atv/turret/buckle_mob(mob/living/M, force = 0)
 	..()
```
Comment (AnturK) [inline code/modules/vehicles/atv.dm]: . = ..()

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 3]
```diff
@@ -524,26 +524,33 @@
...
+	if(riding_datum)
+		riding_datum.restore_position(buckled_mob)
+		. = ..()
 
 /mob/living/simple_animal/user_buckle_mob(mob/living/M, mob/user)
+	if(user.incapacitated())
+		return
+	for(var/atom/movable/A in get_turf(src))
+		if(A.density)
+			if(A != src && A != M)
```
Comment (AnturK) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: You can put these checks in one line

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 4]
```diff
@@ -524,26 +524,33 @@
...
-			return
-		for(var/atom/movable/A in get_turf(src))
-			if(A.density)
-				if(A != src && A != M)
-					return
-		M.loc = get_turf(src)
-		riding_datum.handle_vehicle_offsets()
 		riding_datum.ridden = src
+		riding_datum.handle_vehicle_offsets()
+	..()
```
Comment (AnturK) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: . = ..()

**Comment by Militaires** [comment]
im gonna freeze this and if bm speedwagon goes ill keep it that way

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 5]
```diff
@@ -553,7 +551,6 @@
 		riding_datum.handle_vehicle_offsets()
 
 
-/mob/living/simple_animal/buckle_mob()
+/mob/living/simple_animal/buckle_mob(mob/living/buckled_mob, force = 0, check_loc = 1)
 	..()
```
Comment (AnturK) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: . = ..()

**Pull request merged**